test-lib.sh

```diff
@@ -322,40 +349,54 @@ EOF
 # Uses: $IMAGE_NAME - name of the image being tested
 function ct_check_exec_env_vars() {
   local tmpdir exec_envs cid old_IFS env_filter
-  local var_name stripped filtered_envs
+  local var_name stripped filtered_envs loop_envs
   env_filter=${1:-"^X_SCLS=\|/opt/rh\|/opt/app-root"}
   tmpdir=$(mktemp -d)
   CID_FILE_DIR=${CID_FILE_DIR:-$(mktemp -d)}
   # Get environment variables from `docker run`
-  docker run --rm "$IMAGE_NAME" /bin/bash -c "env" | sort > "$tmpdir/run_envs"
+  loop_envs=$(docker run --rm "$IMAGE_NAME" /bin/bash -c "env")
   # Get environment variables from `docker exec`
   ct_create_container "test_exec_envs" bash -c "sleep 1000" >/dev/null
   cid=$(ct_get_cid "test_exec_envs")
-  docker exec "$cid" env | sort > "$tmpdir/exec_envs"
+  exec_envs=$(docker exec "$cid" env)
   # Filter out variables we are not interested in
   # Always check X_SCLS, ignore PWD
   # Check variables from `docker run` that have alternative paths inside (/opt/rh, /opt/app-root)
-  exec_envs=$(cat "$tmpdir/exec_envs")
-  while read -r variable; do
-    var_name=$(echo "$variable" | awk -F= '{ print $1 }')
-    stripped=$(echo "$variable" | awk -F= '{ print $2 }')
-    filtered_envs=$(echo "$exec_envs" | grep "^$var_name=")
-    [ -z "$filtered_envs" ] && { echo "$var_name not found during \` docker exec\`"; return 1; }
-    old_IFS=$IFS
-    # For each such variable compare its content with the `docker exec` result, use `:` as delimiter
-    IFS=:
-    for value in $stripped; do
-        if [ -n "${filtered_envs##*$value*}" ]; then
-            echo " Value $value is missing from variable $var_name"
-            return 1
-        fi
-    done
-  done <<< "$(grep "$env_filter" "$tmpdir/run_envs" | grep -v "^PWD=")"
-  IFS=$old_IFS
+  ct_check_envs_set "$env_filter" "$exec_envs" "$loop_envs" "*VALUE*" || return 1
   echo " All values present in \`docker exec\`"
   return 0
 }
 
+# ct_check_scl_enable_vars [env_filter]
+# --------------------
+# Checks if all relevant environment variables from `docker run`
+# are set twice after a second call of `scl enable $SCLS`.
+# Argument: env_filter - optional string passed to grep used for
+#   choosing which variables to check in the test case.
+#   Defaults to paths containing enabled SCLS in the image
+# Uses: $IMAGE_NAME - name of the image being tested
+function ct_check_scl_enable_vars() {
+  local tmpdir exec_envs cid old_IFS env_filter enabled_scls
+  local var_name stripped filtered_envs loop_envs
+  env_filter=$1
+  tmpdir=$(mktemp -d)
+  enabled_scls=$(docker run --rm "$IMAGE_NAME" /bin/bash -c "echo \$X_SCLS")
+  if [ -z "$env_filter" ]; then
+    for scl in $enabled_scls; do
+      [ -z "$env_filter" ] && env_filter=/$scl && continue
```

Comment:
And this can be overwritten by:
env_filter is set to empty and defined before `for` cycle
```bash
 env_filter="$env_filter|/$scl"
```

Reply:
You would end up with eg. `|/rh-nodejs10|/rh-postgresql10` this way.